Userland/Libraries/LibDSP/Track.cpp

```diff
@@ -92,6 +96,25 @@ void Track::current_signal(FixedArray<Sample>& output_signal)
     VERIFY(output_signal.size() == source_signal->get<FixedArray<Sample>>().size());
     // The last processor is the fixed mastering processor. This can write directly to the output data. We also just trust this processor that it does the right thing :^)
     m_track_mastering->process_to_fixed_array(*source_signal, output_signal);
+
+    bool false_variable = false;
+    if (m_sample_lock.compare_exchange_strong(false_variable, true)) {
+        for (size_t i = 0; i < m_cached_sample_buffer.size(); i++) {
+            m_cached_sample_buffer[i] = output_signal[i];
+        }
+    }
+}
+
+void Track::cached_signal(FixedArray<Sample>& output_signal)
```

Comment:
You can use a Span for the output signal, that way the user of the API can supply a multitude of containers.